src/heroku/orb.yml

```diff
@@ -1,35 +1,67 @@
 version: 2
 description: "Provides basic functionality for deploying to Heroku"
 
-commands:
-  install:
+executors:
+  default:
+    description:
+      Uses the basic buildpack-deps image, which has the
+      prerequisites for installing heroku's CLI.
     parameters:
-      os:
-        default: “linux”
-        description: The target Operating System for the heroku binary. Must be one of "linux", “darwin”, “win32”.
+      tag:
         type: string
-        # enum types are not yet implemented.
-        # type: enum
-        # enum: ["linux", “darwin”, “win32”]
-      arch:
-        type: string
-        default: "x86"
-        description: The target CPU architecture for the heroku binary. Must be one of “x64”, “x86”, “arm”.
-        # enum ypes are not yet implemented
-        # type: enum
-        # enum: [“x64”, “x86”, “arm”]
+        default: "bionic"
+    docker:
+      - image: buildpack-deps:<< parameters.tag >>
 
+commands:
+  install:
     steps:
       - run:
           name: "Install Heroku CLI, if necessary"
-          # TODO: the assumption of wget, etc. below likely wouldn't work in all cases, especially windows.
           command: |
             if [[ $(command -v heroku) == "" ]]; then
-              wget https://cli-assets.heroku.com/heroku-cli/channels/stable/heroku-cli-{{parameters.os}}-{{parameters.arch}}.tar.gz -O heroku.tar.gz
-              mkdir $HOME/heroku
-              tar -xvzf heroku.tar.gz -C $HOME/heroku --strip-components 1
-              cd $HOME/heroku
-              ./install
+              curl https://cli-assets.heroku.com/install.sh | sh
             else
               echo "Heroku is already installed. No operation was performed."
             fi
+  deploy:
+    parameters:
+      app-name:
+        description:
+          "The name of your Heroku App.
+          For backwards compatibility the literal value
+          `$HEROKU_APP_NAME` is the default, so you can
+          easily use this command by setting an environment
+          variable called HEROKU_APP_NAME"
+        type: string
+        default: $HEROKU_APP_NAME
+      api-key:
+        description:
+          "The API key to use. defaulting to the literal value $HEROKU_API_KEY, so you can
+          populate the environment variable HEROKU_API_KEY and not need to declare a value in code."
+        type: string
+        default: $HEROKU_API_KEY
+      branch:
```

Comment:
Branch filtering should be done at the workflow definition level - let’s remove branch filtering from the job.

Reply:
In this case the primary audience is people moving from 1.0 or onboarding very basic apps. This way they can choose to put a filter here in a single line without needing to understand workflows at first. The default behavior is a noop, so this is innocuous.